packages/graphql/src/schema/resolvers/root-connection.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GraphQLResolveInfo } from "graphql";
+import { InputTypeComposer, SchemaComposer } from "graphql-compose";
+import { PageInfo } from "graphql-relay";
+import { execute } from "../../utils";
+import { translateRead } from "../../translate";
+import { Node } from "../../classes";
+import { Context } from "../../types";
+import getNeo4jResolveTree from "../../utils/get-neo4j-resolve-tree";
+import { isNeoInt } from "../../utils/utils";
+import { createConnectionWithEdgeProperties } from "../pagination";
+
+export default function rootConnectionResolver({ node, composer }: { node: Node; composer: SchemaComposer }) {
+    async function resolve(_root: any, _args: any, _context: unknown, info: GraphQLResolveInfo) {
+        const context = _context as Context;
+        const resolveTree = getNeo4jResolveTree(info);
+
+        const edgeTree = resolveTree.fieldsByTypeName[`${node.name}Connection`].edges;
+        const nodeTree = edgeTree.fieldsByTypeName[`${node.name}Edge`].node;
+
+        context.resolveTree = { ...nodeTree, args: resolveTree.args };
+        context.isRootConnectionField = true;
```

Comment:
I realise we go against this ourselves by setting `context.resolveTree` this way, but we should avoid modifying the context during execution. This feels like it might be better placed as an argument on the `translateRead` function?